image/image.go

```diff
@@ -0,0 +1,32 @@
+package image
+
+import (
+	"errors"
+
+	"github.com/google/go-containerregistry/pkg/name"
+)
+
+func ByRegistry(registry string, images []string) (string, error) {
+	if len(images) < 1 {
+		return "", errors.New("no images provided to search")
+	}
+
+	for _, i := range images {
+		reg, err := ParseRegistry(i)
+		if err != nil {
+			continue
+		}
+		if registry == reg {
+			return i, nil
```

Comment:
This is a minor thing but I was confused the first time I glanced over this code b/c the variable name `i` led me to assume it was referring to the array index.